### PR TITLE
Honor DEFAULT_PAGING_NUM for API

### PR DESCRIPTION
**Pull request by CirnoT**
For some reason it was hardcoded to 10

### CODE REVIEW

**Review by zeripath — APPROVED** [review]

**Review by lafriks — APPROVED** [review]

**Review by techknowlogick — APPROVED** [review]
LGTM, some tests are now failing though

**Comment by CirnoT** [comment]
Should we get rid of tests like `RepositoriesDefaultMax10` or make them dynamic somehow? The option is configurable so we would need to read app.ini first or force some default during test phase?

**Comment by techknowlogick** [comment]
Yeah, or instead of getting rid of them have it test the settings variable.

**Comment by CirnoT** [comment]
Testing the setting variable won't work because it might be larger than amount of repos during test in which case we'll pass regardless whether the default works or is jut too big to get into effect.

**Review by 6543 — APPROVED** [review]

File: integrations/api_repo_test.go [hunk 1]
```diff
@@ -57,6 +58,12 @@ func TestAPISearchRepo(t *testing.T) {
 	user4 := models.AssertExistsAndLoadBean(t, &models.User{ID: 20}).(*models.User)
 	orgUser := models.AssertExistsAndLoadBean(t, &models.User{ID: 17}).(*models.User)
 
+	oldAPIDefaultNum := setting.API.DefaultPagingNum
+	defer func() {
+		setting.API.DefaultPagingNum = oldAPIDefaultNum
+	}()
+	setting.API.DefaultPagingNum = 10
```
Comment (6543) [inline integrations/api_repo_test.go:65]: instead of setting it via variable, why not just add the query arguments?

Comment (CirnoT) [reply integrations/api_repo_test.go:65]: The point is to test default

**Comment by 6543** [comment]
@CirnoT `&page=1&limit=10`

File: integrations/api_repo_test.go (hunk 1 above)
Comment (CirnoT) [reply integrations/api_repo_test.go:65]: If you expand diff you'll see that it tests with query parameter above too

**Comment by techknowlogick** [comment]
ping LG-TM

**Comment by techknowlogick** [comment]
@CirnoT please send backport :)

**Pull request merged**